### ISSUE TITLE
I'm not sure if it's a data issue, please take a look

### ISSUE THREAD

**Issue by pengmingjun08** [stothard-group/cgview-js#12]
![20231026-194403](https://github.com/stothard-group/cgview-js/assets/56865100/4fb8ed8f-eab5-4b79-96fb-d4d60dc0aed3)
![20231026-194359](https://github.com/stothard-group/cgview-js/assets/56865100/c196ed13-c9c7-45cc-a63e-ddd9beea81c0)
Keep scrolling the mouse in the direction of the red arrow to zoom in, and misalignment will occur.
-----------------
In addition, I would like to ask, after I modify the sequence and redraw it, how can I maintain the current zoom ratio and movement position? Can you provide some ideas? Thanks

**Comment by pengmingjun08** [comment]
[testdata.json](https://github.com/stothard-group/cgview-js/files/13176940/testdata.json)

**Comment by sciguy** [comment]
This drawing issue has been fixed in b7ca139e.

As for maintaining the zoom/bp position. Try using the following methods (where `cgv` is your `Viewer` instance):

```js
let bp = cgv.canvas.bpForCanvasCenter()
let zoomLevel = cgv.zoomLevel
// Do your modifications and then move back to the same position
cgv.zoomTo(bp, zoomLevel)
```

**Comment by pengmingjun08** [comment]
> This drawing issue has been fixed in [b7ca139](https://github.com/stothard-group/cgview-js/commit/b7ca139e0ba5a565c49b73e0b60cafa0f5567597).
> 
> As for maintaining the zoom/bp position. Try using the following methods (where `cgv` is your `Viewer` instance):
> 
> ```js
> let bp = cgv.canvas.bpForCanvasCenter()
> let zoomLevel = cgv.zoomLevel
> // Do your modifications and then move back to the same position
> cgv.zoomTo(bp, zoomLevel)
> ```

谢谢，thank you
